src/Apply.ts

```diff
@@ -246,3 +246,107 @@ export function getSemigroup<F, A>(F: Apply<F>, S: Semigroup<A>): () => Semigrou
+    F,
+    U,
+    L,
+    [A, B, C, D]
+  >
+  <A, B, C, D, E>(
+    a: Type3<F, U, L, A>,
+    b: Type3<F, U, L, B>,
+    c: Type3<F, U, L, C>,
+    d: Type3<F, U, L, D>,
+    e: Type3<F, U, L, E>
+  ): Type3<F, U, L, [A, B, C, D, E]>
+}
+export interface SequenceT2<F extends URIS2> {
+  <L, A>(a: Type2<F, L, A>): Type2<F, L, [A]>
+  <L, A, B>(a: Type2<F, L, A>, b: Type2<F, L, B>): Type2<F, L, [A, B]>
+  <L, A, B, C>(a: Type2<F, L, A>, b: Type2<F, L, B>, c: Type2<F, L, C>): Type2<F, L, [A, B, C]>
+  <L, A, B, C, D>(a: Type2<F, L, A>, b: Type2<F, L, B>, c: Type2<F, L, C>, d: Type2<F, L, D>): Type2<F, L, [A, B, C, D]>
+  <L, A, B, C, D, E>(
+    a: Type2<F, L, A>,
+    b: Type2<F, L, B>,
+    c: Type2<F, L, C>,
+    d: Type2<F, L, D>,
+    e: Type2<F, L, E>
+  ): Type2<F, L, [A, B, C, D, E]>
+}
+export interface SequenceT2C<F extends URIS2, L> {
+  <A>(a: Type2<F, L, A>): Type2<F, L, [A]>
+  <A, B>(a: Type2<F, L, A>, b: Type2<F, L, B>): Type2<F, L, [A, B]>
+  <A, B, C>(a: Type2<F, L, A>, b: Type2<F, L, B>, c: Type2<F, L, C>): Type2<F, L, [A, B, C]>
+  <A, B, C, D>(a: Type2<F, L, A>, b: Type2<F, L, B>, c: Type2<F, L, C>, d: Type2<F, L, D>): Type2<F, L, [A, B, C, D]>
+  <A, B, C, D, E>(a: Type2<F, L, A>, b: Type2<F, L, B>, c: Type2<F, L, C>, d: Type2<F, L, D>, e: Type2<F, L, E>): Type2<
+    F,
+    L,
+    [A, B, C, D, E]
+  >
+}
+export interface SequenceT1<F extends URIS> {
+  <A>(a: Type<F, A>): Type<F, [A]>
+  <A, B>(a: Type<F, A>, b: Type<F, B>): Type<F, [A, B]>
+  <A, B, C>(a: Type<F, A>, b: Type<F, B>, c: Type<F, C>): Type<F, [A, B, C]>
+  <A, B, C, D>(a: Type<F, A>, b: Type<F, B>, c: Type<F, C>, d: Type<F, D>): Type<F, [A, B, C, D]>
+  <A, B, C, D, E>(a: Type<F, A>, b: Type<F, B>, c: Type<F, C>, d: Type<F, D>, e: Type<F, E>): Type<F, [A, B, C, D, E]>
+}
+export interface SequenceT<F> {
+  <A>(a: HKT<F, A>): HKT<F, [A]>
+  <A, B>(a: HKT<F, A>, b: HKT<F, B>): HKT<F, [A, B]>
+  <A, B, C>(a: HKT<F, A>, b: HKT<F, B>, c: HKT<F, C>): HKT<F, [A, B, C]>
+  <A, B, C, D>(a: HKT<F, A>, b: HKT<F, B>, c: HKT<F, C>, d: HKT<F, D>): HKT<F, [A, B, C, D]>
+  <A, B, C, D, E>(a: HKT<F, A>, b: HKT<F, B>, c: HKT<F, C>, d: HKT<F, D>, e: HKT<F, E>): HKT<F, [A, B, C, D, E]>
+}
+
+const tupleConstructors: { [key: string]: Function1<any, any> } = {}
+
+/**
+ * Tuple sequencing, i.e., take a tuple of of monadic actions and do them from left-to-right, returning the resulting tuple.
+ * @function
+ * @since 1.5.0
+ * @param F - {@link Apply} instance
+ */
+export function sequenceT<F extends URIS3, U, L>(F: Apply3<F>): SequenceT3<F>
+export function sequenceT<F extends URIS3, U, L>(F: Apply3C<F, U, L>): SequenceT3C<F, U, L>
+export function sequenceT<F extends URIS2>(F: Apply2<F>): SequenceT2<F>
+export function sequenceT<F extends URIS2, L>(F: Apply2C<F, L>): SequenceT2C<F, L>
+export function sequenceT<F extends URIS>(F: Apply1<F>): SequenceT1<F>
+export function sequenceT<F>(F: Apply<F>): SequenceT<F> {
```

Comment:
We must also export the base signature here

```ts
export function sequenceT<F>(F: Apply<F>): SequenceT<F>
```

Reply:
You mean to duplicate it? ok, done